tutorials/polkadot-sdk/parachains/build-custom-pallet/pallet-unit-testing.md

```diff
@@ -0,0 +1,165 @@
+
+As demonstrated in the previous tutorial, the pallet calls to be tested are as follows:
+
+???- "Custom pallet calls"
+
+    ```rust
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/pallet-unit-testing/pallet-calls.rs'
+    ```
+
+The following sub-sections outline various scenarios in which the `custom-pallet` can be tested. Feel free to add these snippets to your `tests.rs` while you read the examples.
+
+### Successful Operations
+
+Verify that the counter can be successfully incremented under normal conditions, ensuring the increment works and the correct event is emitted.
+
+```rust
+--8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/pallet-unit-testing/tests.rs:42:60'
+```
+
+### Preventing Value Overflow
+
+Test that the pallet prevents incrementing beyond the maximum allowed value, protecting against unintended state changes.
+
+```rust
+--8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/pallet-unit-testing/tests.rs:61:75'
+```
+
+### Origin and Access Control
+
+Confirm that sensitive operations like setting counter value are restricted to authorized origins, preventing unauthorized modifications.
+
+```rust
+--8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/pallet-unit-testing/tests.rs:16:28'
+```
+
+### Edge Case Handling
+
+Ensure the pallet gracefully handles edge cases, such as preventing increment operations that would cause overflow.
+
+```rust
+--8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/pallet-unit-testing/tests.rs:76:90'
+```
+
+### Verifying State Changes
+
+Test that pallet operations modify the internal state correctly and maintain expected storage values across different interactions.
+
+```rust
+--8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/pallet-unit-testing/tests.rs:125:141'
+```
+
+### Full Test Suite
+
+You can check the complete `tests.rs` implementation for the `custom pallet` here:
+
+???- "Complete `tests.rs`"
+
+    ```rust
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/pallet-unit-testing/tests.rs'
+    ```
+
+
+## Running Tests
+
+Execute the test suite for your custom pallet using Cargo's test command. This will run all defined test cases and provide detailed output about the test results.
+
+```bash
+cargo test --package custom-pallet
+```
+
+After running the test suite, you should see the following output in your terminal:
+
+--8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/pallet-unit-testing/output.html'
+
+## Where to Go Next
+
+Expand your Polkadot SDK development skills by exploring the [`Pallet Benchmarking`](/tutorials/polkadot-sdk/parachains/build-custom-pallet/pallet-benchmarking){target=\_blank} tutorial.
```

Comment:
This should be changed to a card. Similar to what was done in this PR: https://github.com/polkadot-developers/polkadot-docs/pull/230/files

Reply:
agree!

Reply:
This will be taken care of in [0xlucca/update-tutorial-section](https://github.com/polkadot-developers/polkadot-docs/tree/0xlucca/update-tutorial-section)